### PR TITLE
IGNITE-20236 Get rid of DistributionZonesConfigurationSchema#defaultDataStorage

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-20236

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 1]
```diff
@@ -402,7 +402,8 @@ public CompletableFuture<Integer> createZone(DistributionZoneConfigurationParame
                         }
 
                         if (distributionZoneCfg.dataStorageChangeConsumer() == null) {
-                            zoneChange.changeDataStorage(ch -> ch.convert(zonesConfiguration.defaultDataStorage().value()));
+                            // TODO: IGNITE-20114 Delete when switching to the catalog
+                            zoneChange.changeDataStorage(ch -> ch.convert("aipersist"));
```
Comment (rpuch) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:406]: Let's hardcode it in a constant, not in a literal :)

File: modules/storage-api/src/main/java/org/apache/ignite/internal/storage/DataStorageManager.java [hunk 2]
```diff
@@ -91,28 +67,10 @@ public void stop() throws Exception {
...
-     * Returns the default data storage.
-     *
-     * <p>{@link DistributionZonesConfiguration#defaultDataStorage} is used.
-     */
-    public String defaultDataStorage() {
-        return defaultDataStorageConfig.value();
+    /** Returns the default data storage. */
+    // TODO: IGNITE-20237 Make it configurable
+    public static String defaultDataStorage() {
+        return "aipersist";
```
Comment (rpuch) [inline modules/storage-api/src/main/java/org/apache/ignite/internal/storage/DataStorageManager.java:73]: Let's use a constant

Comment (tkalkirill) [reply modules/storage-api/src/main/java/org/apache/ignite/internal/storage/DataStorageManager.java:73]: Fix it

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java (hunk 1 above)
Comment (tkalkirill) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:406]: That code in the neighboring [PR](https://github.com/apache/ignite-3/pull/2415) disappears, I don’t think that a constant is needed here.

**Review by rpuch — APPROVED** [review]

**Pull request merged**